lib/api/request_adapter.rb

```diff
@@ -25,6 +25,12 @@ def action
                   end
     end
 
+    # currently only custom option actions are returned
+    # default options of :create and :update are not handled
+    def option_action
+      @params["option_action"]
+    end
```

Comment:
not thrilled with this.

It is nice to have this only defined in one spot.
To support the validation check, we would either need to expose `@req.params` or this field.
Seemed the better of the two options.